### DISCUSSION TITLE
Parcel@v2 clean dist plugin

### DISCUSSION THREAD

**Discussion by sergey-shpak** [parcel-bundler/parcel#5285] [General]
Currently I'm using npm scripts to clean my dist directory before each build/watch/serve run, and just curious, - is it doable with some kind of parcel@v2 plugin? Moreover, what kind of plugin is most suitable for cleaning dist directory?

**Comment by sergey-shpak** [comment]
Since all the parcel@v2 plugins can be easily [extended](https://github.com/parcel-bundler/parcel/blob/v2/packages/core/plugin/src/PluginAPI.js) it is possible to implement plugin hooks, and then something like becomes possible:
```
//.parcelrc

// trigger hook on any plugin method
'@parcel/bundler-default.onBundle': 'npm run clean'
```

**Comment by mischnic** [comment]
In theory, probably a reporter (though that runs asynchronously...).

Have you seen this issue? https://github.com/parcel-bundler/parcel/issues/3591#issuecomment-539567824

**Comment by sergey-shpak** [comment]
@mischnic, no I haven't, thanks I'll take a look. 'Reporters' cannot clean dist because there is a possibility to clean all the bundled files
